Remove the outdated example and suggest to use events (#1761)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/Hooks/Hooks/Index.rst

.. note::

   :php:`$GLOBALS['TYPO3_CONF_VARS']['EXTCONF']` was the recommended place where to
   put hook configurations inside third party extensions. It is not recommended anymore

>    to introduce news hooks. Consider to create a :ref:`PSR-14 event <EventDispatcher>` 
>    instead.

Reviewer: ```suggestion
   to introduce news hooks. :ref:`Events <EventDispatcher>` should be used instead.
```
